tests/unit/is_access_granted_ipv4.pl

```diff
@@ -0,0 +1,214 @@
+
+$want{"198.51.100.4/30"}{$undef}{$undef} = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{$undef}{"12"}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{$undef}{"80"}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{"me"}{$undef}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{"me"}{"12"}     = 'OK';
+$want{"198.51.100.4/30"}{"me"}{"80"}     = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{"not"}{$undef}  = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{"not"}{"12"}    = 'KO_ACCESS_DENIED';
+$want{"198.51.100.4/30"}{"not"}{"80"}    = 'KO_ACCESS_DENIED';
+
+$want{"198.51.100.20/30"}{$undef}{$undef} = 'KO_ACCESS_DENIED';
+$want{"198.51.100.20/30"}{$undef}{"12"}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.20/30"}{$undef}{"80"}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.20/30"}{"me"}{$undef}   = 'OK';
+$want{"198.51.100.20/30"}{"me"}{"12"}     = 'OK';
+$want{"198.51.100.20/30"}{"me"}{"80"}     = 'OK';
+$want{"198.51.100.20/30"}{"not"}{$undef}  = 'KO_ACCESS_DENIED';
+$want{"198.51.100.20/30"}{"not"}{"12"}    = 'KO_ACCESS_DENIED';
+$want{"198.51.100.20/30"}{"not"}{"80"}    = 'KO_ACCESS_DENIED';
+
+$want{"198.51.100.36/30"}{$undef}{$undef} = 'KO_ACCESS_DENIED';
+$want{"198.51.100.36/30"}{$undef}{"12"}   = 'OK';
+$want{"198.51.100.36/30"}{$undef}{"80"}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.36/30"}{"me"}{$undef}   = 'KO_ACCESS_DENIED';
+$want{"198.51.100.36/30"}{"me"}{"12"}     = 'OK';
+$want{"198.51.100.36/30"}{"me"}{"80"}     = 'KO_ACCESS_DENIED';
+$want{"198.51.100.36/30"}{"not"}{$undef}  = 'KO_ACCESS_DENIED';
+$want{"198.51.100.36/30"}{"not"}{"12"}    = 'OK';
+$want{"198.51.100.36/30"}{"not"}{"80"}    = 'KO_ACCESS_DENIED';
+
+$want{"198.51.100.52/30"}{$undef}{$undef} = 'OK';
+$want{"198.51.100.52/30"}{$undef}{"12"}   = 'OK';
+$want{"198.51.100.52/30"}{$undef}{"80"}   = 'OK';
+$want{"198.51.100.52/30"}{"me"}{$undef}   = 'OK';
+$want{"198.51.100.52/30"}{"me"}{"12"}     = 'OK';
+$want{"198.51.100.52/30"}{"me"}{"80"}     = 'OK';
+$want{"198.51.100.52/30"}{"not"}{$undef}  = 'OK';
+$want{"198.51.100.52/30"}{"not"}{"12"}    = 'OK';
+$want{"198.51.100.52/30"}{"not"}{"80"}    = 'OK';
+
+$want{"192.0.2.255"}{$undef}{$undef} = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{$undef}{"12"}   = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{$undef}{"80"}   = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"me"}{$undef}   = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"me"}{"12"}     = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"me"}{"80"}     = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"not"}{$undef}  = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"not"}{"12"}    = 'KO_ACCESS_DENIED';
+$want{"192.0.2.255"}{"not"}{"80"}    = 'KO_ACCESS_DENIED';
+
+foreach my $ip (
+    qw{
+    192.0.2.12
+    198.51.100.4
+    192.0.2.13
+    198.51.100.19
+    192.0.2.22
+    198.51.100.35
+    192.0.2.23
+    198.51.100.49
+    198.51.100.4/30
+    198.51.100.20/30
+    198.51.100.36/30
+    198.51.100.52/30
+    192.0.2.255
+    }
+  )
+{
+
+    foreach my $user (undef, "me", "not") {
```

Comment:
You could generate `subtest`s

Reply:
Will use it if at some point there is more than 1 test in the inner loop, thanks for the suggestion!